src/test/resources/lib/configFile/initIDProvider-test.js

```diff
@@ -0,0 +1,159 @@
+const test = require('/lib/xp/testing');
+
+
+
+//////////////////////  Mock workaround (test.mock only works once)
+
+const configFileMocks = {};
+test.mock("/lib/configFile/configFile.js", {
+    getAllIdProviderNames: () => configFileMocks.getAllIdProviderNames(),
+    shouldAutoInit: () => configFileMocks.shouldAutoInit(),
+    getConfigForIdProvider: (idProviderName) => configFileMocks.getConfigForIdProvider(idProviderName)
+});
```

Comment:
This is not possible to do inside the test methods? 
This seems like an unneeded abstraction.

Reply:
It should work, but it doesn't really. Each mock only applies once, for some reason. So "natively", different scenarios can't be tested with different mocks AFAICS (which defies the purpose of mocks), so this workaround was needed.